### PR TITLE
chore(cli): rewrite program.cs for add hostservice and dependency inj…

### PR DESCRIPTION
**Pull request by ng-julien**
Rewrite program.cs for add hostservice and dependency injction.
This will gently allow end-to-end testing and write unit test with mock

### CODE REVIEW

File: src/GUI/efcpt/Services/PackageService.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using ErikEJ.EfCorePowerTools.Options;
+
+namespace ErikEJ.EfCorePowerTools.Services;
+
+internal sealed class PackageService
+{
+    public PackageService(NuGetOptions options)
+    {
+
```
Comment (ErikEJ) [inline src/GUI/efcpt/Services/PackageService.cs:9]: Is something missing here?

File: src/GUI/efcpt/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Debug",
+      "System": "Information",
+      "Microsoft": "Information"
+    }
+  },
+  "NuGet": {
+    "RepositoryUri": "https://api.nuget.org/v3/index.json",
```
Comment (ErikEJ) [inline src/GUI/efcpt/appsettings.json:10]: Why do we need this?

File: src/GUI/efcpt/appsettings.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "PackageKey": "ErikEJ.EFCorePowerTools.CLI"
```
Comment (ErikEJ) [inline src/GUI/efcpt/appsettings.json:11]: What is this for? Also, the name is: ....Cli (casing)

**Comment by ErikEJ** [comment]
Amazing! Just a few commentst/questions

**Review by ErikEJ — CHANGES_REQUESTED** [review]
LGTM - just a few comments

File: src/GUI/efcpt/Services/PackageService.cs (hunk 1 above)
Comment (ng-julien) [reply src/GUI/efcpt/Services/PackageService.cs]: oups!!! my last commit is not up to date

File: src/GUI/efcpt/appsettings.json (hunk 2 above)
Comment (ng-julien) [reply src/GUI/efcpt/appsettings.json]: > Why do we need this?

I suppress this

File: src/GUI/efcpt/appsettings.json (hunk 3 above)
Comment (ng-julien) [reply src/GUI/efcpt/appsettings.json]: > What is this for? Also, the name is: ....Cli (casing)

I suppress this

File: src/GUI/efcpt/Services/PackageService.cs (hunk 1 above)
Comment (ng-julien) [reply src/GUI/efcpt/Services/PackageService.cs]: > Is something missing here?

oups!!! my last commit is not up to date

File: src/GUI/efcpt.6/efcpt.6.csproj [hunk 4]
```diff
@@ -42,8 +56,10 @@
   <ItemGroup>
     <PackageReference Include="Ben.Demystifier" Version="0.4.1" />
     <PackageReference Include="CommandLineParser" Version="2.9.1" />
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="7.0.1" />
```
Comment (ErikEJ) [inline src/GUI/efcpt.6/efcpt.6.csproj:59]: Prefer a 6.0 version here

**Comment by ErikEJ** [comment]
Build error on the EF 6 version (csproj needs updated)

CSC : error CS2001: Source file 'C:\projects\efcorepowertools\src\GUI\efcpt.6\..\efcpt\UpdateChecker.cs' could not be found. [C:\projects\efcorepowertools\src\GUI\efcpt.6\efcpt.6.csproj::TargetFramework=net6.0]

**Comment by ng-julien** [comment]
> Build error on the EF 6 version (csproj needs updated)
> 
> CSC : error CS2001: Source file 'C:\projects\efcorepowertools\src\GUI\efcpt.6..\efcpt\UpdateChecker.cs' could not be found. [C:\projects\efcorepowertools\src\GUI\efcpt.6\efcpt.6.csproj::TargetFramework=net6.0]

done

File: src/GUI/efcpt.6/efcpt.6.csproj (hunk 4 above)
Comment (ng-julien) [reply src/GUI/efcpt.6/efcpt.6.csproj]: > Prefer a 6.0 version here

done

**Comment by ErikEJ** [comment]
Thanks a lot! Highly appreciated

**Pull request merged**